From 56677887688a72abc5b6cfabf363068fea2745af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Mi=C5=82ecki?= <rafal@milecki.pl>
Date: Thu, 16 Mar 2023 20:12:53 +0530
Subject: [PATCH 170/500] OpenWrt:
 930-Revert-Revert-Revert-driver-core-Set-fw_devlink-on-b.patch
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Revert "Revert "Revert "driver core: Set fw_devlink=on by default"""

This reverts commit ea718c699055c8566eb64432388a04974c43b2ea.

With of_platform_populate() called for MTD partitions that commit breaks
probing devices which reference MTD in device tree.

Link: https://lore.kernel.org/all/696cb2da-20b9-b3dd-46d9-de4bf91a1506@gmail.com/T/#u
Signed-off-by: Rafał Miłecki <rafal@milecki.pl>

Change-Id: I6950f73bcf47149b09882be3934a92303098abf6
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
Signed-off-by: Saravanan Jaganathan <quic_sjaganat@quicinc.com>
(cherry picked from commit dac3f5c64fe1244b1ea8dcab8ae717dd4846f303)
---
 drivers/base/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/base/core.c b/drivers/base/core.c
index 4d8b315c48a1..a0bca8df335c 100644
--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -1640,7 +1640,7 @@ static void device_links_purge(struct device *dev)
 #define FW_DEVLINK_FLAGS_RPM		(FW_DEVLINK_FLAGS_ON | \
 					 DL_FLAG_PM_RUNTIME)
 
-static u32 fw_devlink_flags = FW_DEVLINK_FLAGS_ON;
+static u32 fw_devlink_flags = FW_DEVLINK_FLAGS_PERMISSIVE;
 static int __init fw_devlink_setup(char *arg)
 {
 	if (!arg)
-- 
2.34.1

